sql/core/src/test/scala/org/apache/spark/sql/execution/metric/SQLMetricsSuite.scala

```diff
@@ -570,4 +572,33 @@ class SQLMetricsSuite extends SparkFunSuite with SQLMetricsTestUtils with Shared
       }
     }
   }
+
+  test("InputMetrics---bytesRead") {
```

Comment:
This isn't really testing the code you changed. It's replicating something similar and testing that. I don't think this test helps. Ideally you would write a test for any path that uses `FileScanRDD` and check its metrics. Are there tests around here that you could 'piggyback' onto? maybe an existing test of the metrics involving `ColumnarBatch ` than can be changed to trigger this case.

It may be hard, I don't know. Worth looking to see if there's an easy way to test this.

Reply:
It is too hard，the test needs  involve `ColumnarBatch`,
in addition, we must capture the `bytesRead` in the process of execution, not the task end.